specification/sections/01-flag-evaluation.md

```diff
@@ -401,3 +401,73 @@ The global API object might expose a `shutdown` function, which will call the re
 Alternatively, implementations might leverage language idioms such as auto-disposable interfaces or some means of cancellation signal propagation to allow for graceful shutdown.
 
 see: [`shutdown`](./02-providers.md#25-shutdown)
+
+### 1.7. Provider Lifecycle Management
+
+The implementation maintains an internal representation of the state of configured providers, tracking the lifecycle of each provider.
+This state of the provider is exposed on associated `clients`.
+
+The diagram below illustrates the possible states and transitions of the `state` field for a provider during the provider lifecycle.
+
+```mermaid
+---
+title: Provider State
+---
+stateDiagram-v2
+    direction LR
+    [*] --> NOT_READY
+    NOT_READY --> READY:initialize()
+    NOT_READY --> ERROR:initialize()
+    READY --> ERROR*
+    ERROR --> READY*
+    READY --> STALE*
+    STALE --> READY*
+    STALE --> ERROR*
+    READY --> NOT_READY:shutdown()
+    STALE --> NOT_READY:shutdown()
+    ERROR --> NOT_READY:shutdown()
+```
+
+\* transitions occurring when associated events are spontaneously emitted from the provider
+
+Note that SDKs implementing the [static context (client-side) paradigm](../glossary.md#static-context-paradigm) define additional states to facilitate [context reconciliation](./02-providers.md#26-provider-context-reconciliation).
```

Comment:
````suggestion
```mermaid
stateDiagram-v2
    direction LR
    [*] --> NOT_READY
    NOT_READY --> READY:initialize()
    NOT_READY --> ERROR:initialize()
    READY --> ERROR:*
    ERROR --> READY:*
    READY --> STALE:*
    STALE --> READY:*
    STALE --> ERROR:*
    READY --> NOT_READY:shutdown()
    STALE --> NOT_READY:shutdown()
    ERROR --> NOT_READY:shutdown()
    READY --> CONTEXT_PENDING:::optional:setContext()
    CONTEXT_PENDING --> READY:*

    classDef optional fill:#999
```

\* transitions occurring when associated events are spontaneously emitted from the provider

Note that only SDKs implementing the [static context (client-side) paradigm](../glossary.md#static-context-paradigm) define `CONTEXT_PENDING` to facilitate [context reconciliation](./02-providers.md#26-provider-context-reconciliation).
````

I would not make all the `STATE*` states a different state. In the rendered diagram they look like separate, final states. I changed it so the * is on the edges, what do you think @toddbaert?

Additionally I feel like it would be good to have the `CONTEXT_PENDING` so I added it in another color.

The change rendered looks like this: 
```mermaid
stateDiagram-v2
    direction LR
    [*] --> NOT_READY
    NOT_READY --> READY:initialize()
    NOT_READY --> ERROR:initialize()
    READY --> ERROR:*
    ERROR --> READY:*
    READY --> STALE:*
    STALE --> READY:*
    STALE --> ERROR:*
    READY --> NOT_READY:shutdown()
    STALE --> NOT_READY:shutdown()
    ERROR --> NOT_READY:shutdown()
    READY --> CONTEXT_PENDING:::optional:setContext()
    CONTEXT_PENDING --> READY:*

    classDef optional fill:#999
```

Reply:
Oh, I didn't notice that, nice catch. It's just because I added the * at the last moment and it rendered as a different state. Accidental!

Reply:
> Additionally I feel like it would be good to have the CONTEXT_PENDING so I added it in another color. This way we have all information at one place here.

Love it!!!